src/HotChocolate/Diagnostics/src/Diagnostics/ActivityEnricher.cs

```diff
@@ -582,33 +587,35 @@ void BuildPath()
 
     protected virtual void EnrichError(IError error, Activity activity)
     {
-        var tags = new List<KeyValuePair<string, object?>>
+        if (error.Exception is { } exception)
+        {
+            activity.RecordException(exception);
+        }
+
+        var tags = new ActivityTagsCollection
         {
-            new("graphql.error.message", error.Message),
-            new("graphql.error.code", error.Code),
+            new(AttributeExceptionMessage, error.Message),
+            new(AttributeExceptionType, error.Code)
         };
 
         if (error.Locations is { Count: > 0, })
         {
-            if (error.Locations.Count == 1)
+            for (var i = 0; i < error.Locations.Count; i++)
             {
-                tags.Add(new("graphql.error.location.column", error.Locations[0].Column));
-                tags.Add(new("graphql.error.location.line", error.Locations[0].Line));
-            }
-            else
-            {
-                for (var i = 0; i < error.Locations.Count; i++)
-                {
-                    tags.Add(new($"graphql.error.location[{i}].column", error.Locations[i].Column));
-                    tags.Add(new($"graphql.error.location[{i}].line", error.Locations[i].Line));
-                }
+                tags["graphql.error.location.column"] = error.Locations[i].Column;
+                tags["graphql.error.location.line"] = error.Locations[i].Line;
+
+                activity.AddEvent(new ActivityEvent(AttributeExceptionEventName, default, tags));
             }
         }
-
-        activity.AddEvent(new("Error", tags: new(tags)));
+        else
+        {
+            activity.AddEvent(new ActivityEvent(AttributeExceptionEventName, default, tags));
+        }
     }
```

Comment:
@michaelstaib

If an `IError` has an exception, we now report the exception regularly. We also report the GraphQL Error as an "exception" to make it easier to trace them.

As the tags of an event should be static, we should not use `graphql.error.location[{i}].column`. With this PR, we currently report each location as its own error. It's unclear if this is the best approach, as it could lead to confusion when an error with multiple locations is reported as two separate errors.

Perhaps we should only report the first location to avoid this potential confusion.